### DISCUSSION TITLE
can consumer shard mode  be used with  AckCumulative

### DISCUSSION THREAD

**Discussion by someview** [apache/pulsar#22148] [Q&A]
can consumer shard mode  be used with  AckCumulative， multi comsumer receive message, if one consumer use AckCumulative(pulsar-client-go) to a msg,  other consumer msg that msg id is lower than this is also acked??

**Comment by jak78** [comment]
Hi @someview! Using the Shared subscription mode, you don't have an ordered delivery guarantee so the cumulative acknowledge would not make sense here. Cumulatively acknowledging messages makes more sense with Exclusive and Failover subscriptions. Could you please share your use case and what you are trying to achieve?
